pkg/serverconfig/metrics.go

```diff
@@ -19,27 +58,189 @@ const (
+	"netobserv-plugin":                PluginVendorRedHat,
+	"nmstate-console-plugin":          PluginVendorRedHat,
+	"node-remediation-console-plugin": PluginVendorRedHat,
+	"odf-console":                     PluginVendorRedHat,
+	"odf-multicluster-console":        PluginVendorRedHat,
+
+	// Unchanged template name from https://github.com/openshift/console-plugin-template
+	"console-plugin-template": PluginVendorDemo,
+	"my-plugin":               PluginVendorDemo,
+}
+
+type PluginState string
+
+const (
+	// existing ConsolePlugin resource, and part of the console config
+	PluginStateEnabled PluginState = "enabled"
+	// existing ConsolePlugin resource, but not part of the console config
+	PluginStateDisabled PluginState = "disabled"
+	// configured console config, but there is no ConsolePlugin resource
+	PluginStateNotFound PluginState = "notfound"
+)
+
 func (m *Metrics) GetCollectors() []prometheus.Collector {
 	return []prometheus.Collector{
-		m.pluginsInfo,
 		m.perspectivesInfo,
+		m.pluginsInfo,
+	}
+}
+
+func (m *Metrics) MonitorPlugins(
+	userSettingsClient *http.Client,
+	userSettingsEndpoint string,
+	serviceAccountToken string,
+) {
+	go func() {
+		time.Sleep(3 * time.Second)
+		go m.updatePluginMetric(userSettingsClient, userSettingsEndpoint, serviceAccountToken)
+	}()
+
+	ticker := time.NewTicker(updateConsolePluginInterval)
+	quit := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				m.updatePluginMetric(userSettingsClient, userSettingsEndpoint, serviceAccountToken)
+			case <-quit:
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+}
+
+func (m *Metrics) updatePluginMetric(
+	k8sClient *http.Client,
+	k8sEndpoint string,
+	serviceAccountToken string,
+) {
+	klog.Info("serverconfig.Metrics: Update ConsolePlugin metrics...\n")
+	startTime := time.Now()
+
+	// Create a new plugin info map that is based on the last report to report also removed ConsolePlugins.
+	pluginInfo := make(map[PluginVendor]map[PluginState]int)
+	for lastPluginVendor, lastPluginStates := range m.lastPluginInfo {
+		for lastPluginState, _ := range lastPluginStates {
+			if pluginInfo[lastPluginVendor] == nil {
+				pluginInfo[lastPluginVendor] = make(map[PluginState]int)
+			}
+			pluginInfo[lastPluginVendor][lastPluginState] = 0
+		}
+	}
+
+	ctx := context.TODO()
+	config := &rest.Config{
+		Transport:   k8sClient.Transport,
+		Host:        k8sEndpoint,
+		BearerToken: serviceAccountToken,
+	}
+	client, err := dynamic.NewForConfig(config)
+	if err != nil {
+		klog.Errorf("serverconfig.Metrics: Failed to create service account client: %v\n", err)
+	}
+
+	var consolePlugins *unstructured.UnstructuredList
+	if client != nil {
+		consolePlugins, err = client.Resource(consolePluginResource).List(ctx, v1.ListOptions{})
+		if err != nil {
+			klog.Errorf("serverconfig.Metrics: Failed to get all installed ConsolePlugins: %v\n", err)
+		}
+	}
+	consolePluginNames := make(map[string]bool)
```

Comment:
Also assembling the `consolePluginNames` would be good to have in a separate method/helper.
Would be easier to unit-test